Internal catch error

## Issue description

I get an internal catch error when I try to install.

## Reproducible example code

```bash
$ cmake .. -DCMAKE_INSTALL_PREFIX:PATH=~/opt
$ make install                                                                                                                                                                              [master]
Scanning dependencies of target pybind11_cross_module_tests
[  2%] Building CXX object tests/CMakeFiles/pybind11_cross_module_tests.dir/pybind11_cross_module_tests.cpp.o
[  4%] Linking CXX shared module ../../tests/pybind11_cross_module_tests.cpython-36m-x86_64-linux-gnu.so
[  4%] Built target pybind11_cross_module_tests
Scanning dependencies of target pybind11_tests
[  7%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/pybind11_tests.cpp.o
[  9%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_buffers.cpp.o
[ 11%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_builtin_casters.cpp.o
[ 14%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_call_policies.cpp.o
[ 16%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_callbacks.cpp.o
[ 19%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_chrono.cpp.o
[ 21%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_class.cpp.o
[ 23%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_constants_and_functions.cpp.o
[ 26%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_copy_move.cpp.o
[ 28%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_docstring_options.cpp.o
[ 30%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_eigen.cpp.o
[ 33%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_enum.cpp.o
[ 35%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_eval.cpp.o
[ 38%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_exceptions.cpp.o
[ 40%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_factory_constructors.cpp.o
[ 42%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_iostream.cpp.o
[ 45%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_kwargs_and_defaults.cpp.o
[ 47%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_local_bindings.cpp.o
[ 50%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_methods_and_attributes.cpp.o
[ 52%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_modules.cpp.o
[ 54%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_multiple_inheritance.cpp.o
[ 57%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_numpy_array.cpp.o
[ 59%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_numpy_dtypes.cpp.o
[ 61%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_numpy_vectorize.cpp.o
[ 64%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_opaque_types.cpp.o
[ 66%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_operator_overloading.cpp.o
[ 69%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_pickling.cpp.o
[ 71%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_pytypes.cpp.o
[ 73%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_sequences_and_iterators.cpp.o
[ 76%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_smart_ptr.cpp.o
[ 78%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_stl.cpp.o
[ 80%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_stl_binders.cpp.o
[ 83%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_tagbased_polymorphic.cpp.o
[ 85%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_virtual_functions.cpp.o
[ 88%] Linking CXX shared module ../../tests/pybind11_tests.cpython-36m-x86_64-linux-gnu.so
------ pybind11_tests.cpython-36m-x86_64-linux-gnu.so file size: 1888496
[ 88%] Built target pybind11_tests
Scanning dependencies of target external_module
[ 90%] Building CXX object tests/test_embed/CMakeFiles/external_module.dir/external_module.cpp.o
[ 92%] Linking CXX shared module ../../../tests/test_embed/external_module.cpython-36m-x86_64-linux-gnu.so
[ 92%] Built target external_module
Scanning dependencies of target test_embed
[ 95%] Building CXX object tests/test_embed/CMakeFiles/test_embed.dir/catch.cpp.o
/home/tdegeus/prog/clone/pybind11/tests/test_embed/catch.cpp: In function ‘int main(int, const char**)’:
/home/tdegeus/prog/clone/pybind11/tests/test_embed/catch.cpp:19:50: error: invalid conversion from ‘const char**’ to ‘char**’ [-fpermissive]
     auto result = Catch::Session().run(argc, argv);
                                                  ^
In file included from /home/tdegeus/prog/clone/pybind11/tests/test_embed/catch.cpp:13:0:
/home/tdegeus/opt/include/catch/catch.hpp:9253:9: note:   initializing argument 2 of ‘int Catch::Session::run(int, char**)’
     int Session::run( int argc, char* argv[] ) {
         ^~~~~~~
make[2]: *** [tests/test_embed/CMakeFiles/test_embed.dir/catch.cpp.o] Error 1
make[1]: *** [tests/test_embed/CMakeFiles/test_embed.dir/all] Error 2
make: *** [all] Error 2
```

--- a/tests/test_embed/catch.cpp
+++ b/tests/test_embed/catch.cpp
@@ -14,7 +14,7 @@
 
 namespace py = pybind11;
 
-int main(int argc, const char *argv[]) {
+int main(int argc, char *argv[]) {
     py::scoped_interpreter guard{};
     auto result = Catch::Session().run(argc, argv);
 

